src/Microsoft.AspNet.Mvc.Extensions/FormatterMappings.cs

```diff
@@ -27,7 +27,7 @@ public void SetMediaTypeMappingForFormat([NotNull] string format, [NotNull] Medi
         {
             ValidateContentType(contentType);
             format = RemovePeriodIfPresent(format);
-            _map[format] = MediaTypeHeaderValue.Parse(contentType.ToString());
+            _map[format] = contentType.CopyAsReadOnly();
```

Comment:
FYI...this looks different as here we have a chance to intercept the mediatype and convert to readonly copy before putting in the dictionary...but in case of formatters, the media type header value is directly aded to the list...

To keep it consistent with formatters, i could just leave this line of code to be `_map[format] = contentType;` as we are anyway creating a copy in output formatter, thoughts?